src/util/simplify_expr.cpp

```diff
@@ -553,7 +553,7 @@ simplify_exprt::simplify_typecast(const typecast_exprt &expr)
       to_bitvector_type(expr_type).get_width())
   {
     auto new_expr = expr;
-    new_expr.op() = expr.op().op0();
+    new_expr.op() = to_typecast_expr(expr.op()).op();
```

Comment:
If we're doing this (which I think is a good idea) then we should get rid of the `expr.op().operands().size() == 1 &&` in the condition.

Reply:
done